flink-java/src/main/java/org/apache/flink/api/java/RemoteEnvironment.java

```diff
@@ -1,271 +1,287 @@
+	 * @param clientConfig The configuration used by the client that connects to the cluster.
+	 * @param jarFiles The JAR files with code that needs to be shipped to the cluster. If the program uses
+	 *                 user-defined functions, user-defined input formats, or any libraries, those must be
+	 *                 provided in the JAR files.
+	 * @param globalClasspaths The paths of directories and JAR files that are added to each user code 
+	 *                 classloader on all nodes in the cluster. Note that the paths must specify a 
+	 *                 protocol (e.g. file://) and be accessible on all nodes (e.g. by means of a NFS share).
+	 *                 The protocol must be supported by the {@link java.net.URLClassLoader}.
+	 */
+	public RemoteEnvironment(String host, int port, Configuration clientConfig,
+			String[] jarFiles, URL[] globalClasspaths) {
+		if (!ExecutionEnvironment.areExplicitEnvironmentsAllowed()) {
+			throw new InvalidProgramException(
+					"The RemoteEnvironment cannot be instantiated when running in a pre-defined context " +
+							"(such as Command Line Client, Scala Shell, or TestEnvironment)");
+		}
+		if (host == null) {
+			throw new NullPointerException("Host must not be null.");
+		}
+		if (port < 1 || port >= 0xffff) {
+			throw new IllegalArgumentException("Port out of range");
+		}
+
+		this.host = host;
+		this.port = port;
+		this.clientConfiguration = clientConfig == null ? new Configuration() : clientConfig;
+		if (jarFiles != null) {
+			this.jarFiles = new URL[jarFiles.length];
+			for (int i = 0; i < jarFiles.length; i++) {
+				try {
+					this.jarFiles[i] = new File(jarFiles[i]).getAbsoluteFile().toURI().toURL();
+				} catch (MalformedURLException e) {
+					throw new IllegalArgumentException("JAR file path invalid", e);
+				}
+			}
+		}
+		else {
+			this.jarFiles = null;
+		}
+		this.globalClasspaths = globalClasspaths;
+	}
+
+	// ------------------------------------------------------------------------
+
+	@Override
+	public JobExecutionResult execute(String jobName) throws Exception {
+		ensureExecutorCreated();
+
+		Plan p = createProgramPlan(jobName);
+
+		// Session management is disabled, revert this commit to enable
+		//p.setJobId(jobID);
+		//p.setSessionTimeout(sessionTimeout);
+
+		JobExecutionResult result = executor.executePlan(p);
+
+		this.lastJobExecutionResult = result;
+		return result;
+	}
+
+	@Override
+	public String getExecutionPlan() throws Exception {
+		Plan p = createProgramPlan("plan", false);
+
+		// make sure that we do not start an new executor here
+		// if one runs, fine, or not, we create a local executor (lightweight) and let it
+		// generate the plan
+		if (executor != null) {
+			return executor.getOptimizerPlanAsJSON(p);
+		}
+		else {
+			PlanExecutor le = PlanExecutor.createLocalExecutor(null);
+			return le.getOptimizerPlanAsJSON(p);
+		}
+	}
+
+	@Override
+	public String getSqlExecutionPlan(boolean extended) throws Exception {

```

Comment:
I think you added only this method. 
Can you check why the whole file is marked as changed and revert unrelated changes?